File: src/code-samples/langsmith/smithdb-migration/runs-query-selecting-fields-after.go
// :snippet-start: runs-query-selecting-fields-after-go
// :codegroup-tab: After
package main

import (
	"context"
	"fmt"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name:  langsmith.F("default"),
	Limit: langsmith.F(int64(1)),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
project := sessions.Items[0]

// must explicitly list every field needed; default returns only id
runs, err := client.Runs.QueryV2(ctx, langsmith.RunQueryV2Params{
	ProjectIDs: langsmith.F([]string{project.ID}),
	Selects: langsmith.F([]langsmith.RunQueryV2ParamsSelect{
		langsmith.RunQueryV2ParamsSelectID,
		langsmith.RunQueryV2ParamsSelectName,
		langsmith.RunQueryV2ParamsSelectRunType,
		langsmith.RunQueryV2ParamsSelectStatus,
		langsmith.RunQueryV2ParamsSelectStartTime,
		langsmith.RunQueryV2ParamsSelectInputs,
		langsmith.RunQueryV2ParamsSelectError,
	}),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
for _, run := range runs.Items {
	fmt.Println(run.ID, run.Name, run.RunType, run.Status, run.StartTime, run.Inputs, run.Error)
	// :remove-start:
	break
	// :remove-end:
}
// :remove-start:
}

// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-query-selecting-fields-after.kt
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.2.0
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.11

// :snippet-start: runs-query-selecting-fields-after-kt
// :codegroup-tab: After
import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.runs.RunQueryV2Params
import com.langchain.smith.models.sessions.SessionListParams

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-runs-query-selecting-fields-after] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }

// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

val project = client.sessions().list(
    SessionListParams.builder().name("default").limit(1L).build()
).items().first()
// must explicitly list every field needed; default returns only id
val runs = client.runs().queryV2(
    RunQueryV2Params.builder()
        .addProjectId(project.id())
        .addSelect(RunQueryV2Params.Select.ID)
        .addSelect(RunQueryV2Params.Select.NAME)
        .addSelect(RunQueryV2Params.Select.RUN_TYPE)
        .addSelect(RunQueryV2Params.Select.STATUS)
        .addSelect(RunQueryV2Params.Select.START_TIME)
        .addSelect(RunQueryV2Params.Select.INPUTS)
        .addSelect(RunQueryV2Params.Select.ERROR)
        .build()
).items()
for (run in runs) {
    println("${run.id()} ${run.name()} ${run.runType()} ${run.status()} ${run.startTime()} ${run.inputs()} ${run.error()}")
    // :remove-start:
    break
    // :remove-end:
}
// :remove-start:
}
// :remove-end:
// :snippet-end:
